### PR TITLE
Register WorldGenerarorModifier with new API

### PR DESCRIPTION
**Pull request by NeusFear**
The docs have a depreciated API documented, so this solves that.

### CODE REVIEW

File: source/plugin/wgen/modifiers.rst [hunk 1]
```diff
@@ -70,8 +70,8 @@ the first argument and your modifier as the second.
 .. code-block:: java
 
     @Listener
-    public void onGameInitialization(GameInitializationEvent event) {
-        Sponge.getRegistry().register(WorldGeneratorModifier.class , new MyModifier());
+    public void onWorldRegistration(GameRegistryEvent.Register<WorldGeneratorModifier> event) {
```
Comment (ST-DDT) [inline source/plugin/wgen/modifiers.rst]: I would rather call the method `onWGenModifierRegistration` or `onWorldGeneratorModifierRegistration`.

**Review by ST-DDT — CHANGES_REQUESTED** [review]
A few lines above this code sample there is also a textual reference to `:javadoc:GameRegistry#register(Class, T)`, that needs to be changed as well.

File: source/plugin/wgen/modifiers.rst (hunk 1 above)
Comment (NeusFear) [reply source/plugin/wgen/modifiers.rst]: gotcha, updated.

File: source/plugin/wgen/modifiers.rst [hunk 2]
```diff
@@ -63,15 +63,14 @@ apply your generator changes to nether worlds.
 Registering a WorldGeneratorModifier
 ====================================
 
-Now that you have created our modifier, you need to register it. A good time to do this is during the ``INITIALIZATION``
-State. To register it, simply call :javadoc:`GameRegistry#register(Class, T)` with ``WorldGeneratorModifier.class`` as
-the first argument and your modifier as the second.
+Now that you have created our modifier, you need to register it. A good time to do this is during the ``GAMEREGISTRY``
+State. To register it, simply call :javadoc:`GameRegistryEvent#Register<WorldGeneratorModifier>(T)` with your modifier as the argument.
```
Comment (ST-DDT) [inline source/plugin/wgen/modifiers.rst]: It's not a state. It's an event.
You dont call it. You listen to it and register it there.

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: source/plugin/wgen/modifiers.rst (hunk 2 above)
Comment (ST-DDT) [reply source/plugin/wgen/modifiers.rst]: Also missing javadoc import.

Comment (NeusFear) [reply source/plugin/wgen/modifiers.rst]: I changed my word choice for explaining the event, and I think I fixed the Javadoc import, but I don't exactly know how they work on GitHub, so you may have to commit that change.

File: source/plugin/wgen/modifiers.rst [hunk 3]
```diff
@@ -63,15 +63,14 @@ apply your generator changes to nether worlds.
 Registering a WorldGeneratorModifier
 ====================================
 
-Now that you have created our modifier, you need to register it. A good time to do this is during the ``INITIALIZATION``
-State. To register it, simply call :javadoc:`GameRegistry#register(Class, T)` with ``WorldGeneratorModifier.class`` as
-the first argument and your modifier as the second.
+Now that you have created our modifier, you need to register it. A good time to do this is during the ``GAMEREGISTRY``
+Event. To register it, simply lsiten to :javadoc:`GameRegistryEvent.Register` then register from there.
```
Comment (ST-DDT) [inline source/plugin/wgen/modifiers.rst]: `lsiten` -> `listen`

**Review by ST-DDT — CHANGES_REQUESTED** [review]
The javadoc import has to be added at the top of the file (see the javadoc import block). You can try to build the docs locally or get feedback from travis CI whether you have fixed that bug.

**Review by NeusFear — COMMENTED** [review]
Should work now

**Comment by NeusFear** [comment]
@ST-DDT it checks out now.

**Comment by ST-DDT** [comment]
Sorry, Im busy currently and cannot review it.

File: source/plugin/wgen/modifiers.rst [hunk 4]
```diff
@@ -63,15 +64,14 @@ apply your generator changes to nether worlds.
 Registering a WorldGeneratorModifier
 ====================================
 
-Now that you have created our modifier, you need to register it. A good time to do this is during the ``INITIALIZATION``
-State. To register it, simply call :javadoc:`GameRegistry#register(Class, T)` with ``WorldGeneratorModifier.class`` as
-the first argument and your modifier as the second.
+Now that you have created our modifier, you need to register it. A good time to do this is during the ``GAMEREGISTRY``
+Event. To register it, simply listen to :javadoc:`GameRegistryEvent` then register from there.
```
Comment (ST-DDT) [inline source/plugin/wgen/modifiers.rst]: I would rephrase it like this: (including some additional cross references to other pages)


````text
Now that you have created the modifier, all you have to do is register it in the :javadoc:`GameRegistry`. To do this you need to create a :doc:`listener </plugin/event/listeners>` for the :javadoc:`GameRegistryEvent.Register<WorldGeneratorModifier>` event and register your instance there, as shown in the following example:
````


With that changes you only have to change the javadoc import at the top from `GameRegistryEvent` to `GameRegistryEvent.Register`.

File: source/plugin/wgen/modifiers.rst [hunk 5]
```diff
@@ -6,7 +6,8 @@ WorldGeneratorModifiers
     org.spongepowered.api.CatalogType
     org.spongepowered.api.GameRegistry
     org.spongepowered.api.data.DataContainer
-    org.spongepowered.api.world.gen.WorldGeneratorModifier
+    org.spongepowered.api.event.game.GameRegistryEvent
+    org.spongepowered.api.event.game.WorldGeneratorModifier
```
Comment (ST-DDT) [inline source/plugin/wgen/modifiers.rst]: Wrong import: Should be `org.spongepowered.api.world.gen.WorldGeneratorModifier`

**Review by ST-DDT — CHANGES_REQUESTED** [review]
I tested the code example and the WGM was successfully registered.

**Review by ST-DDT — APPROVED** [review]

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/1842281/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/1842281/plugin/wgen/modifiers.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/769#issuecomment-425225364), not to the direct url to the preview._
